src/pool.c

```diff
@@ -1537,6 +1537,52 @@ rpc_on_block_template(const char* data, rpc_callback_t *callback)
     json_object_put(root);
 }
 
+static int
+startup_scan_round_shares()
+{
+    int rc;
+    char *err;
+    MDB_txn *txn;
+    MDB_cursor *cursor;
+
+    if ((rc = mdb_txn_begin(env, NULL, MDB_RDONLY, &txn)) != 0)
+    {
+        err = mdb_strerror(rc);
+        log_error("%s", err);
+        return rc;
+    }
+    if ((rc = mdb_cursor_open(txn, db_shares, &cursor)) != 0)
+    {
+        err = mdb_strerror(rc);
+        log_error("%s", err);
+        mdb_txn_abort(txn);
+        return rc;
+    }
+    MDB_cursor_op op = MDB_LAST;
+    while (1)
+    {
+        MDB_val key;
+        MDB_val val;
+        rc = mdb_cursor_get(cursor, &key, &val, op);
+        if (rc != 0 && rc != MDB_NOTFOUND)
+        {
+            err = mdb_strerror(rc);
+            log_error("%s", err);
+            break;
+        }
```

Comment:
And if rc==MDB_NOTFOUND ?

Reply:
You need to deal with  MDB_NOTFOUND.

Reply:
You need to deal with  MDB_NOTFOUND.

Reply:
Removed by accident along with MDB_PREV_DUP.
Reverted.